pkg/sync/github/issue.go

```diff
@@ -0,0 +1,132 @@
+package github
+
+import (
+	"bytes"
+	"crypto/md5"
+	"fmt"
+	"hash"
+	"io"
+	"time"
+
+	"github.com/gofrs/uuid"
+	"github.com/naggie/dstask"
+)
+
+// IssueData is a compact way to represent an issue
+// so that templates can be expanded simply (without nested properties)
+type IssueData struct {
+
+	// internal properties
+	uuidHash hash.Hash     // to generate UUID's
+	buf      *bytes.Buffer // to expand templates into
+
+	// populated from our hash
+	uuid uuid.UUID
+	UUID string
+
+	// populated from our scraping config
+	User string
+	Repo string
+
+	// populated from the data GitHub returned to us
+	Author    string
+	Body      string
+	ClosedAt  time.Time
+	Closed    bool
+	CreatedAt time.Time
+	Milestone string
+	Number    int
+	State     string
+	Title     string
+	Url       string
+}
+
+func NewIssueData() *IssueData {
+	return &IssueData{
+		// to write key issue features into, to generate the UUID
+		uuidHash: md5.New(),
+		buf:      &bytes.Buffer{},
+	}
+}
+
+// Init sets all properties to match the given user, repo and Github data
+func (id *IssueData) Init(user, repo string, i Issue) {
```

Comment:
Does this need to be a method? Is there any reason we cannot pass these as params to `NewIssueData`?

Reply:
this way we can reuse the hash and id buffer, by retaining a single IssueData instance and reusing it for different issues.